core/genesis_test.go

```diff
@@ -176,3 +175,51 @@ func TestSetupGenesis(t *testing.T) {
 		}
 	}
 }
+
+// TestRegistryInGenesis tests if the params.RegistrySmartContract that defined in the genesis block sits in the blockchain
+func TestRegistryInGenesis(t *testing.T) {
```

Comment:
love that we're doing a tests for this. Still, i would also ask that when we launch a node with a genesis block we should check that registry is there.

Not sure if we should do this on `geth init` or on the startup logic when we run `geth`. Maybe in both.

Basically, we want to make sure that if you've done `geth init` then `geth` you have the registry installed.

Reply:
Added `geth` and `geth init` tests

Reply:
@YoduYodu maybe i didn't express myself correctly. The idea here is not to add new tests, but to check that genesis has the contract deployed whenever we run `geth` and `geth init`. So, not a test, but a check you do during normal execution, just to be sure you're not operating with a genesis without it.

For example, imagine i create a genesis.json with an empty alloc. then i do `geth init mybogusgenesis.json` and then `geth`

With current code, you will only notice something is wrong when you reach to the place we did the check (codeSize != 0), and now we don't. So it would fail and it would be hard to debug.

Instead, what i'm asking is taht when you run geth init with a bogus genesis.json it fails